cardano-mempool/src/Cardano/Memory/Pool.hs

```diff
@@ -0,0 +1,217 @@
+{-# LANGUAGE BangPatterns #-}
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE KindSignatures #-}
+{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE MagicHash #-}
+{-# LANGUAGE RankNTypes #-}
+{-# LANGUAGE RecordWildCards #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE UnboxedTuples #-}
+
+module Cardano.Memory.Pool where
```

Comment:
Could use a little bit more documentation: a summary of what this module is about and how the lock-free algorithm works, and some explanations / summaries on at least the functions and types that are intended as the API for this module (`Block`, `Pool`, `initPool`, `grabNextPoolForeignPtr`).

I would also consider explicitly exporting only the intended API, though I understand it may be necessary to expose the rest for testing purposes, and an `Internal.hs` setup isn't everyone's idea of elegance.

Reply:
:100:

Reply:
Done. No need for internal module, since internal functions aren't testable in isolation